### PR TITLE
feat: Adds Whitelist to Series Progress Algorithm

### PR DESCRIPTION
**Pull request by redreceipt**
This will let us specify which series channels we want to look through for progress items. So instead of including all sermon series _and_ devotional series by default, we can pick one.

### CODE REVIEW

**Comment by redreceipt** [comment]
@vinnyjth not sure what is going on with tests here but what do you think about this approach? it uses the blacklist/whitelist paradigm but I didn't know if you wanted to get away from that as stated in one of the comments.

**Comment by vinnyjth** [comment]
@redreceipt Those two new queries scare me a lil bit, do you think we could consolidate that `eq channel id` / `ne channel id` logic into the call to `getFromIds`? It returns a RequestBuilder object so you might be able to chain off of that.

**Comment by redreceipt** [comment]
let me see. I like that better too

**Comment by redreceipt** [comment]
something like this?

File: packages/apollos-data-connector-rock/src/content-items/data-source.js [hunk 1]
```diff
@@ -553,11 +546,19 @@ export default class ContentItem extends RockApolloDataSource {
...
+    // only search through allowed channels
+    channelIds.forEach((id) => {
+      cursor = cursor.andFilter(`ContentChannelId eq ${id}`);
+    });
+
+    // exclude campaign channels
+    ROCK_MAPPINGS.CAMPAIGN_CHANNEL_IDS.forEach((id) => {
+      cursor = cursor.andFilter(`ContentChannelId ne ${id}`);
+    });
+    return cursor;
```
Comment (redreceipt) [inline packages/apollos-data-connector-rock/src/content-items/data-source.js:561]:
```suggestion

    return cursor;
```

**Comment by redreceipt** [comment]
alright ready now

**Comment by vinnyjth** [comment]
Well done 👏

**Review by vinnyjth — APPROVED** [review]

**Pull request merged**